eng/generators.targets

```diff
@@ -1,45 +1,66 @@
-<Project InitialTargets="ConfigureGenerators">
+<Project>
+
+  <PropertyGroup>
```

Comment:
Has there been an official discussion about this model for enabling generators? This file is named in a generic fashion so I assume the idea is other future generators should be included here. Do we need doc or a sign off for this pattern?

Reply:
There has not, but this is the first generator that we're enabling across the repo, so we can define the pattern.

I came up with this mechanism after chatting with @ViktorHofer offline to make sure that it is static-graph-compatible (the previous model was not compatible with the NuGet Restore model we use to keep build perf reasonable).

Reply:
@ViktorHofer insight was what I was alluding to. If this pattern was discussed with him I am good to go - :shipit: